### ISSUE TITLE
'docker pull' doesn't cleanup downloaded layers if disrupted before return

### ISSUE THREAD

**Issue by maplain** [vmware/vic#4762]
**User Statement:**

As a docker user, I would like to have docker remove the downloaded layers if I stop the 'docker pull' before it finishes.

**Details:**
'docker pull' will remove downloaded layers if it receives a sigint, but vch returns without cleaning up already downloaded layers.

**Acceptance Criteria:**
when using docker through vch, docker pull will remove downloaded layers before return if it's interrupted.

#For bug reports, please include the information below:

**VIC version:**
v1.1.0-rc2-10000-72eee33

**Deployment details:**

bin/vic-machine-darwin create --force --target xxx --password xxx --thumbprint  --name dev --user xxx  --image-store=datastore1 --volume-store=datastore1/dev:default --compute-resource cls --bridge-network DPortGroup --no-tlsverify --no-tls

**Steps to reproduce:**
1. docker -H xxx pull redis
2. after several layers show 'pull complete', hit Ctrl-c;
3. docker -H xxx pull redis again, those layers will show 'Already exists'

**Expected behavior:**
when docker -H xxx pull redis again, those layers will be pulled again.

**Comment by hickeng** [comment]
Confirmed that regular docker behaviour has changed to the described semantic:
```
WARNING: No swap limit support
vagrant@devbox:~/vic$ docker pull redis
Using default tag: latest
latest: Pulling from library/redis
6d827a3ef358: Already exists
787f13ab8ea9: Pull complete
14faf2bdac07: Pull complete
d52905ba81f3: Pull complete
de079db59e0e: Extracting [===========================>                       ]  3.015MB/5.471MB
34450ffcb858: Download complete
9a08f1c245e8: Download complete
^C
vagrant@devbox:~/vic$
vagrant@devbox:~/vic$
vagrant@devbox:~/vic$
vagrant@devbox:~/vic$
vagrant@devbox:~/vic$
vagrant@devbox:~/vic$ docker pull redis
Using default tag: latest
latest: Pulling from library/redis
6d827a3ef358: Already exists
787f13ab8ea9: Pull complete
14faf2bdac07: Pull complete
d52905ba81f3: Pull complete
de079db59e0e: Extracting [==========================>                        ]  2.949MB/5.471MB
34450ffcb858: Download complete
9a08f1c245e8: Download complete
^C
```

@maplain Do you want to have the layers removed to avoid dangling layers in the case the image is never used, or for some other reason? The layers that we report already exist are valid and up-to-date given they're referenced by a content based digest (standard docker image/registry.v2)

**Comment by maplain** [comment]
@hickeng  yes, I think it would be nice to remove them if user disrupt pulling for some reason. This feels more logically consistent.
